types/wlr_buffer.c

```diff
@@ -219,14 +193,15 @@ struct wlr_client_buffer *wlr_client_buffer_import(
 			wlr_drm_buffer_from_resource(resource);
 		buffer = wlr_buffer_lock(&drm_buffer->base);
 	} else {
-		wlr_log(WLR_ERROR, "Cannot upload texture: unknown buffer type");
-
-		// Instead of just logging the error, also disconnect the client with a
-		// fatal protocol error so that it's clear something went wrong.
-		wl_resource_post_error(resource, 0, "unknown buffer type");
+		wlr_log(WLR_ERROR, "Unknown buffer type");
 		return NULL;
 	}
+	
+	return buffer;
+}
 
+struct wlr_client_buffer *wlr_client_buffer_create(struct wlr_buffer *buffer,
+		struct wlr_renderer *renderer, struct wl_resource *resource) {
 	struct wlr_texture *texture = wlr_texture_from_buffer(renderer, buffer);
 	wlr_buffer_unlock(buffer);
```

Comment:
This `wlr_buffer_unlock` call makes `wlr_client_buffer_create` a bit weird: it decrements ref'count of the `wlr_buffer` passed in. This is surprising for callers, I think it'd be better to let the callers do it if they need to.

Reply:
Removed it